doc/bsp/raspberrypi.md

```diff
@@ -0,0 +1,330 @@
+# Raspberry Pi
+
+EMLinux for Raspberry Pi is __not__ officlally supported. However Raspberry Pi is popular device so it would be nice to test EMLinux on it.
+This document describe how to setup EMLinux for Raspberry Pi 3 model B+.
+
+## Directory structures
+
+In this document, following directory structure is used.
+
+```
+build/
+downloads/
+emlinux/repos/meta-debian 
+emlinux/repos/meta-debian-extended
+emlinux/repos/meta-emlinux
+emlinux/repos/poky
+firmware
+linux-firmware
+u-boot
+```
+
+## Build EMLinux
+
+1. Basic setup
+
+Run following command.
+
+```
+$ source repos/meta-emlinux/scripts/setup-emlinux build
+```
+
+2. Add layer
+
+EMLinux for Raspberry Pi uses additional layer, so add it.
+
+```
+$ bitbake-layers add-layer /<path to emlinux directory>/repos/meta-emlinux/bsp/meta-raspberrypi_emlinux
+```
+
+3. Build image
+
+In the build directroy, you can build image. You needs to set raspberrypi3-64 to MACHINE variable. 
+
+```
+$ MACHINE=raspberrypi3-64 bitbake core-image-minimal
+```
+
+## Build u-boot
+
+Building u-boot, you must have cross compiler. 
+
+1. Download source code
+
+```
+$ git clone https://github.com/u-boot/u-boot
+```
+
+2. Checkout release version
+
+```
+$ cd u-boot
+$ git checkout -b v201907 v2019.07
+```
+
+3. Build u-boot
+
+```
+$ make CROSS_COMPILE=aarch64-linux-gnu- rpi_3_defconfig
+$ make CROSS_COMPILE=aarch64-linux-gnu- u-boot.bin
+```
+
+## Create sdcard image
+
+This document uses parted to create partions and /dev/sde as sdcard device.
+
+1. Create fat32 partition for boot.
+
+```
+$ sudo parted /dev/sde mkpart primary fat32 1024KiB 65MiB Information: You may need to update /etc/fstab.
```

Comment:
Is this correct? What is `Information: ... `

Reply:
an, before the ``` Information ~ ``` line should new line( '\n'). I'll remove it.